File: src/main/java/com/project/pantau/mapper/ReportStatusMapper.java
package com.project.pantau.mapper;

import com.project.pantau.dto.report_status.ReportStatusResponse;
import com.project.pantau.entity.ReportStatusHistory;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

import java.util.List;

@Mapper(config = MapStructConfig.class)
public interface ReportStatusMapper {
    @Mapping(target = "actorRole", source = "actor.role")
    ReportStatusResponse toResponse(ReportStatusHistory entity);

    List<ReportStatusResponse> toResponse(List<ReportStatusHistory> entities);
}

File: backend/internal/dto/mapper/report_status_mapper.go
package mapper

import (
	reportstatus "pantau/internal/dto/report_status"
	"pantau/internal/entity"
)

func ReportStatusToResponse(history *entity.ReportStatusHistory) *reportstatus.ReportStatusResponse {
	if history == nil {
		return nil
	}
	return &reportstatus.ReportStatusResponse{
		ID:         history.ID,
		FromStatus: history.FromStatus,
		ToStatus:   history.ToStatus,
		Note:       history.Note,
		ActorRole:  history.Actor.Role,
		CreatedAt:  history.CreatedAt,
	}
}

func ReportStatusesToResponse(histories []entity.ReportStatusHistory) []reportstatus.ReportStatusResponse {
	if histories == nil {
		return nil
	}
	responses := make([]reportstatus.ReportStatusResponse, len(histories))
	for i := range histories {
		responses[i] = *ReportStatusToResponse(&histories[i])
	}
	return responses
}
